designs/2019-enable-es2017-by-default/README.md

```diff
@@ -0,0 +1,138 @@
+- Start Date: 2019-02-28
+- RFC PR: https://github.com/eslint/rfcs/pull/16
+- Authors: Teddy Katz
+
+# Enable ES2017 parsing and globals by default
+
+## Summary
+
+This RFC proposes to:
+
+* Add an `es2017` environment which enables all ES2015 and ES2017 globals.
+* Enable the `es2017` environment by default.
+* Change the `es6` environment to no longer enable `parserOptions: { ecmaVersion: 6 }`.
```

Comment:
I'm also not a fan of this. Once again, we may be breaking a large number of existing shareable configs. If people only have `es6: true`, then their configs will start throwing parsing errors with an upgrade.

Reply:
maybe a better option than breaking env, is adding a new top level option that wouldn’t imply an ecmaVersion?

Reply:
> If people only have `es6: true`, then their configs will start throwing parsing errors with an upgrade.

This doesn't seem like it will be an issue if we change the default `parserOptions` value at the same time. (I have a note about this under "Backwards Compatibility Analysis".)

* Users who enable the `es6` env without also configuring `parserOptions` will end up with the new default parsing behavior of `parserOptions: { ecmaVersion: 2017 }`, so no new parsing errors will be produced.
* Users who enable the `es6` env and also configure `parserOptions` will still have their own parser options used, because `parserOptions` have precedence over `env`.

Reply:
This feels a bit like trying to pull one dish out of a stack of dishes and hoping the food on top doesn't spill.

And as I pointed out in the previous bullet, I don't think implicitly changing what people think they are configuring is a good idea because it will end up breaking a large number of users without much additional benefit, especially if we add an `es2017` environment that doesn't include `parserOptions` and includes the additional globals.